pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -818,3 +843,101 @@ func getInitContainerImage(scanSpec *complianceoperatorv1alpha1.ComplianceScanSp
 	logger.Info("Content image", "image", image)
 	return image
 }
+
+func createAggregatorPodName(scanName string) string {
+	return dnsLengthName("aggregator-pod-", "aggregator-pod-%s", scanName)
+}
+
+func newAggregatorPod(scanInstance *complianceoperatorv1alpha1.ComplianceScan, logger logr.Logger) *corev1.Pod {
+	podName := createAggregatorPodName(scanInstance.Name)
+	logger.Info("Creating an aggregator pod for scan", "pod", podName, "scan", scanInstance.Name)
+
+	podLabels := map[string]string{
+		"complianceScan": scanInstance.Name,
+	}
+
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: scanInstance.Namespace,
+			Labels:    podLabels,
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: "compliance-operator",
+			InitContainers: []corev1.Container{
+				{
+					Name:  "content-container",
+					Image: getInitContainerImage(&scanInstance.Spec, logger),
+					Command: []string{
+						"sh",
+						"-c",
+						"cp /*.xml /content",
+					},
+					ImagePullPolicy: corev1.PullAlways,
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "content-dir",
+							MountPath: "/content",
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "log-collector",
+					Image: GetComponentImage(AGGREGATOR),
+					Args: []string{
+						"--content=" + absContentPath(scanInstance.Spec.Content),
+						"--scan=" + scanInstance.Name,
+					},
+					SecurityContext: &corev1.SecurityContext{
+						Privileged: &trueVal,
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "content-dir",
+							MountPath: "/content",
+						},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
```

Comment:
perhaps this could restart if there's an error?

Reply:
Yes, I was thinking about letting k8s handle errors as well. The reason I left PolicyNever was to not let k8s restart the pod indefinitely in case there was an unrecoverable error.

In the end we should think about what kinds of unrecoverable errors we might have. Of course that could be bugs (can't find the scan because it's mislabeled, wrong arguments passed to the aggregator), we can't do much about those and if we had a bug and a restartPolicy, then the pod would just loop. Then there are issues like can't create remediations, can't read the files. We could make them recoverable by logging an error and exiting the container gracefully (`os.Exit(0)`) so that the pod is spawned again. Then finally there is the case where remediations differ. We could also handle this in the aggregator by setting the scan status to done/error ourselves, but then I'm not sure if we want multiple components (not just the scan controller) to update the scan status.

tl;dr I'm not sure either, I'd welcome another opinion.

Reply:
I think a more relevant thing is: can we detect if we have an unrecoverable error? If so, lets always restart, and if we detect that things are just not moving forward... we kill the pod and surface the error on the CR's status.

The status should only be updated by the controllers. If we get the workloads updating status, we would really get into a mess.

Reply:
Actually, could we just always restart the pod and in the controller abort the scan if the pod reaches CrashLoopBackoff?

Reply:
that sounds reasonable.